server/list.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+type list struct {
+	listID string
+	userID string
+	p      *Plugin
+}
+
+const (
+	// MyListKey is the key used to store the order of the owned todos
+	MyListKey = ""
+	// InListKey is the key used to store the order of received todos
+	InListKey = "_in"
+	// OutListKey is the key used to store the order of sent todos
+	OutListKey = "_out"
+)
+
+func (p *Plugin) getInListForUser(userID string) *list {
+	return &list{
+		listID: InListKey,
+		userID: userID,
+		p:      p,
```

Comment:
In addition to the above comments:

1. Once you refactor, you shouldn't need to put the plugin in the list struct. Generally, think: does the list struct need to know about the plugin? I would say no.
1. The list manager will do all the getting, adding, removing, and have `getItems` and `getItemsOrdered` functions. 

We'll work on more specifics once you've gone through some refactoring. :)